### PR TITLE
Remove shutdown marker file when shutting down

### PR DESCRIPTION
**Pull request by MichelHollands**
**What this PR does / why we need it**:

This removes the shutdown marker file when actually shutting down. If the PVC is re-used then the marker file shouldn't be there.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -547,6 +547,17 @@ func (i *Ingester) stopping(_ error) error {
 	// we need to mark the ingester service as "failed", so Loki will shut down entirely.
 	// The module manager logs the failure `modules.ErrStopProcess` in a special way.
 	if i.terminateOnShutdown && errs.Err() == nil {
+		shutdownMarkerPath := path.Join(i.cfg.ShutdownMarkerPath, shutdownMarkerFilename)
```
Comment (DylanGuedes) [inline pkg/ingester/ingester.go:550]: nitpicking: do you mind moving this check to a separate function?

**Review by DylanGuedes — CHANGES_REQUESTED** [review]

File: pkg/ingester/ingester.go (hunk 1 above)
Comment (MichelHollands) [reply pkg/ingester/ingester.go]: No problems. It's moved.

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -547,11 +547,28 @@ func (i *Ingester) stopping(_ error) error {
...
+	if err != nil {
+		level.Error(util_log.Logger).Log("msg", "error checking shutdown marker file exists", "err", err)
+	}
+	if exists {
+		err = removeShutdownMarker(shutdownMarkerPath)
+		if err != nil {
+			level.Error(util_log.Logger).Log("msg", "error removing shutdown marker file", "err", err)
+		}
+	}
+
```
Comment (DylanGuedes) [inline pkg/ingester/ingester.go:569]:
```suggestion
```

**Review by DylanGuedes — APPROVED** [review]
lgtm, just a final nit

**Pull request merged**